src/atlases.jl

```diff
@@ -0,0 +1,251 @@
+
+"""
+    abstract type AbstractAtlas end
+
+An abstract class for atlases.
+"""
+abstract type AbstractAtlas end
+
+"""
+    struct RetractionAtlas{
+        TInvRetr<:AbstractInverseRetractionMethod,
+        TRetr<:AbstractRetractionMethod,
+        TBasis<:AbstractBasis,
+    } <: AbstractAtlas
+
+An atlas indexed by points on a manifold, such that coordinate transformations are performed
+using retractions, inverse retractions and coordinate calculation for a given basis.
+
+# See also
+
+[`AbstractAtlas`](@ref), [`AbstractInverseRetractionMethod`](@ref),
+[`AbstractRetractionMethod`](@ref), [`AbstractBasis`](@ref)
+"""
+struct RetractionAtlas{
+    TInvRetr<:AbstractInverseRetractionMethod,
+    TRetr<:AbstractRetractionMethod,
+    TBasis<:AbstractBasis{<:Any,TangentSpaceType},
+} <: AbstractAtlas
+    invretr::TInvRetr
+    retr::TRetr
+    basis::TBasis
+end
+
+function RetractionAtlas(
+    invretr::AbstractInverseRetractionMethod,
+    retr::AbstractRetractionMethod,
+)
+    return RetractionAtlas(invretr, retr, DefaultOrthonormalBasis())
+end
+RetractionAtlas() = RetractionAtlas(LogarithmicInverseRetraction(), ExponentialRetraction())
+
+get_default_atlas(M::Manifold) = RetractionAtlas()
+
+"""
+    get_point_coordinates(M::Manifold, A::AbstractAtlas, i, p)
+
+Calculate coordinates of point `p` on manifold `M` in chart from an [`AbstractAtlas`](@ref)
+`A` at index `i`.
+
+# See also
+
+[`get_point`](@ref), [`get_chart_index`](@ref)
+"""
+get_point_coordinates(::Manifold, ::AbstractAtlas, ::Any, ::Any)
+
+function get_point_coordinates(M::Manifold, A::AbstractAtlas, i, p)
```

Comment:
Should the decorator machinery be used here at all?

Reply:
Nice observation – indeed, this function could be transparent for metrics as well as groups, since the atlas could be used for the manifold without having any of this in mind.

Reply:
That's an interesting point. `RetractionAtlas` does actually depend on the metric though so maybe it shouldn't always be transparent for metric decorators?

Reply:
So I think `get_default_atlas`, `get_point_coordinates`, `get_point`, `get_chart_index`, `transition_map` and `induced_basis` should be forwarded through the group decorator?

Reply:
Hm, I tried adding some decorator annotations and it just reminded me that I still don't know how it's supposed to work :confused: .